code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,183 @@
+	var/equipment_name = "generic"
+	var/equipment_path = null
+	var/cost = 0
+	var/amount = 0
+
+/datum/GBP_equipment/New(name, path, cost, amount)
+	src.equipment_name = name
+	src.equipment_path = path
+	src.cost = cost
+	src.amount = amount
+
+/obj/machinery/engi_points_manager/Initialize()
+	radio = new(src)
+	radio.listening = 0
+	radio.frequency = 1357
+	..()
+
+/obj/machinery/engi_points_manager/Destroy()
+	if(radio)
+		qdel(radio)
+		radio = null
+	return ..()
+
+
+/obj/machinery/engi_points_manager/power_change()
+	..()
+	update_icon()
+
+/obj/machinery/engi_points_manager/update_icon()
+	if(powered())
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]-off"
+
+/obj/machinery/engi_points_manager/attack_hand(mob/user)
+	if(..())
+		return
+	interact(user)
+
+/obj/machinery/engi_points_manager/interact(mob/user)
+	var/dat
+	dat +="<div class='statusDisplay'>"
+	dat += "You currently have <td>[round(GBP)]</td> engineering voucher points<br>"
+	dat += "You have earned a total of <td>[round(GBPearned)]</td> this shift<br>"
+	dat += "</div>"
+	dat += "<br><b>Equipment point cost list:</b><BR><table border='0' width='300'>"
+	for(var/datum/GBP_equipment/prize in prize_list)
+		dat += "<tr><td>[prize.equipment_name]</td><td>[prize.cost]</td><td><A href='?src=\ref[src];purchase=\ref[prize]'>Purchase</A></td></tr>"
+	dat += "</table>"
+
+	var/datum/browser/popup = new(user, "vending", "Engineering Point Redemption", 400, 350)
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/engi_points_manager/Topic(href, href_list)
+	if(..())
+		return
+	if(href_list["purchase"])
+		var/datum/GBP_equipment/prize = locate(href_list["purchase"])
+		if (!prize || !(prize in prize_list))
+			return
+		if(prize.cost > GBP)
+			return
+		else if(prize.cost <= GBP) // Placeholder spaghetti calm your shit
+			GBP -= prize.cost
+			for(var/i in 1 to prize.amount)
+				for(var/obj/machinery/engi_points_delivery/D in machines)
+					D.icon_state = "geardist-load"
+					playsound(D, 'sound/machines/Ding.ogg', 100, 1)
+					spawn(20)
+						new prize.equipment_path(get_turf(D))
+						D.icon_state = "geardist"
+					if(prize.cost == 20000) // Still a placeholder
+						new /obj/item/clothing/head/helmet/space/chronos(get_turf(src))
+					feedback_add_details("Engi_equipment_bought",
+					"[src.type]|[prize.equipment_path]")
+		else
+			GBP -= prize.cost
+			new prize.equipment_path(src.loc)
+			feedback_add_details("Engi_equipment_bought",
+				"[src.type]|[prize.equipment_path]")
+	updateUsrDialog()
+
+/obj/machinery/engi_points_manager/attackby(obj/item/I, mob/user, params)
+	return ..()
+
+/obj/machinery/engi_points_manager/process()
+	power_export_bonus = 0
+	for(var/obj/machinery/power/exporter/PE in machines)
+		power_export_bonus = PE.drain_rate/200
+	if(GBP_alarm_cooldown <= world.time)
+		var/limit = 0 // ugh, to stop it from checking the Centcom Computer
+		for(var/obj/machinery/computer/station_alert/SA in machines)

```

Comment:
I simply need a better solution here, MSO mentioned a function that would check for the nearest one...